src/dmd/root/filename.d

```diff
@@ -634,6 +634,7 @@ nothrow:
                 auto n = combine(p.toDString, name);
                 if (exists(n))
                     return n;
+                mem.xfree(cast(void*)n.ptr);
```

Comment:
Take a look at the implementation of combine(). It can return `name`, which this line will then free, corrupting memory.

Reply:
Good catch.  Checking for that explicitly now.  Are there plans to try to use RC strings or something now that we have copy constructors?  The current string manip stuff seems easy to screw up

Reply:
I think the plan is to use D strings as much as possible. But those are usually GC allocated and since the GC is disable in DMD, that will cause memory leaks. But that's how most classes are allocated anyway.